src/backend/crew_ai/tasks.py

```
@@ -117,14 +111,41 @@ def _extract_json_by_key(raw: str, required_key: str, log_prefix: str) -> Option
             decoder = json.JSONDecoder()
             data, _ = decoder.raw_decode(raw[brace_start:])
             if isinstance(data, dict) and required_key in data:
-                fixed = json.dumps(data)
-                logger.info(f"✅ Guardrail: Found {log_prefix} at position {brace_start}")
-                return fixed
+                candidates.append({
+                    'data': data,
+                    'position': brace_start,
+                    'value': data.get(required_key, '')
+                })
+                logger.debug(f"Found candidate {log_prefix} at position {brace_start}")
```

Ruff (0.14.14)

[warning] Logging statement uses f-string

(G004)